### PR TITLE
HTTP code status assertions now fail tests

### LINKED ISSUE #379: assert.HTTP(Success|Redirect|Error) do not fail tests; they are inconsistent with other assert.* functions
`HTTPSuccess`, `HTTPRedirect`,  and `HTTPError` do not `Fail` tests even though the [docs](https://godoc.org/github.com/stretchr/testify/require#HTTPError) use the same wording as `HTTPBodyContains` and `HTTPBodyNotContains` which do `Fail` tests.

In order to properly use them now, one must do this:
```
assert.True(t, assert.HTTPError(t, ...))
```

In order to be consistent with the rest of the assert package, they should `Fail` tests so that they are used like:
```
assert.HTTPError(t, ...)
```

### PR DESCRIPTION
**Pull request by medzin**
It resolves #379 by failing tests when assertion is false or is unable to construct HTTP request.

### CODE REVIEW

File: assert/http_assertions_test.go [hunk 1]
```diff
@@ -19,21 +19,52 @@ func httpError(w http.ResponseWriter, r *http.Request) {
...
+func TestHTTPRedirect(t *testing.T) {
+	assert := New(t)
+
+	mockT1 := new(testing.T)
+	assert.Equal(HTTPRedirect(mockT1, httpOK, "GET", "/", nil), false)
+	assert.True(mockT1.Failed())
+
+	mockT2 := new(testing.T)
+	assert.Equal(HTTPRedirect(mockT2, httpRedirect, "GET", "/", nil), true)
+	assert.False(mockT2.Failed(), "WHAT")
```
Comment (ernesto-jimenez) [inline assert/http_assertions_test.go]: WHAT? :)

File: assert/http_assertions.go [hunk 2]
```diff
@@ -26,11 +26,18 @@ func httpCode(handler http.HandlerFunc, method, url string, values url.Values) i
...
+	code, err := httpCode(handler, method, url, values)
+	if err != nil {
+		Fail(t, fmt.Sprintf("Failed to build test request, got error: %s", err))
 		return false
 	}
-	return code >= http.StatusOK && code <= http.StatusPartialContent
+
+	isSuccessCode := code >= http.StatusOK && code <= http.StatusPartialContent
+	if !isSuccessCode {
+		Fail(t, fmt.Sprintf("Expected HTTP success status code for \"%s\" but received \"%d\"", url+"?"+values.Encode(), code))
```
Comment (ernesto-jimenez) [inline assert/http_assertions.go]: I would rather do `"Expected HTTP success status code for %q but received %d"`

**Comment by medzin** [comment]
@ernesto-jimenez fixed!

**Comment by ernesto-jimenez** [comment]
merged, thanks!

**Pull request merged**